includes/Checker/Checks/Plugin_Updater_Check.php

```diff
@@ -0,0 +1,187 @@
+					esc_html( implode( ', ', $matches ) )
+				)
+			);
+		}
+	}
+
+	/**
+	 * Looks for plugin updater code in plugin files and amends the given result with an error if found.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result    The check result to amend, including the plugin context to check.
+	 * @param array        $php_files List of absolute PHP file paths.
+	 */
+	protected function look_for_plugin_updaters( Check_Result $result, array $php_files ) {
+
+		$needles = array(
+			"'plugin-update-checker'",
+			'WP_GitHub_Updater',
+			'WPGitHubUpdater',
+			'#class [A-Z_]+_Plugin_Updater#i',
+			'#updater\.\w+\.\w{2,5}#i',
+			'site_transient_update_plugins',
+		);
+
+		foreach ( $needles as $needle ) {
+			$match = self::file_str_contains( $php_files, $needle );
+			if ( $match ) {
+				$this->add_result_error_for_file(
+					$result,
+					$match,
+					sprintf(
+						/* translators: %s: The match file name. */
+						__( 'Plugin Updater detected. These are not permitted in WordPress.org hosted plugins. Detected: %s', 'plugin-check' ),
+						esc_html( $needle )
+					)
+				);
+			}
+		}
+	}
+
+	/**
+	 * Looks for plugin updater routines in plugin files and amends the given result with an error if found.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result    The check result to amend, including the plugin context to check.
+	 * @param array        $php_files List of absolute PHP file paths.
+	 */
+	protected function look_for_updater_routines( Check_Result $result, array $php_files ) {
+
+		$needles = array(
+			'#pre_set_site_transient_update_\w+#i',
+			'auto_update_plugin',
+			'#_site_transient_update_\w+#i',
+		);
+
+		foreach ( $needles as $needle ) {
+			$match = self::file_str_contains( $php_files, $needle );
+			if ( $match ) {
+				$this->add_result_error_for_file(
+					$result,
+					$match,
+					sprintf(
+						/* translators: %s: The match file name. */
+						__( 'Detected code which may be altering WordPress update routines. Detected: %s', 'plugin-check' ),
+						esc_html( $needle )
+					),
+					false,
+					'update_modification_detected'
+				);
+			}
+		}
+	}
+
+	/**
+	 * Amends the given result with an error for the given obfuscated file and tool name.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Check_Result $result       The check result to amend, including the plugin context to check.
+	 * @param string       $updater_file Absolute path to the updater file found.
+	 * @param string       $message      Error message for updater.
+	 * @param bool         $error        Whether it is an error message. Defaults to true.
+	 * @param string       $code         Violation code according to the message. Default "plugin_updater_detected".
+	 */
+	private function add_result_error_for_file( Check_Result $result, $updater_file, $message, $error = true, $code = 'plugin_updater_detected' ) {
```

Comment:
Maybe it would be better here to keep the parameters in the same order as they are in the `Check_Result::add_message()` method? Additionally, I would advise against using a default value for `$code`: Either it should always be passed, or it should always be the same value for the method, since otherwise it can become confusing.
```suggestion
	private function add_result_error_for_file( Check_Result $result, $error, $message, $code, $updater_file ) {
```

Reply:
Addressed in https://github.com/10up/plugin-check/pull/189/commits/e1572f3ffd07a9aeba3463cc7b981cdc2b386ead